jxl/src/render/stages/noise.rs

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// TODO(zond): remove once we use this!
+#![allow(dead_code, unused_variables)]
+
+use crate::{
+    features::noise::Noise,
+    render::{RenderPipelineInOutStage, RenderPipelineInPlaceStage, RenderPipelineStage},
+};
+
+pub struct ConvolveNoiseStage {
```

Comment:
Since this operates independently on each channel, wouldn't it make more sense to have a stage that operates on a channel at a time? (I am aware this is different from what libjxl does :-))

Reply:
I have no opinion about this. I suggest we leave it like this for now and let anyone interested refactor it.

Reply:
sure, can you leave a TODO then?

Reply:
OK, adding the TODO was ~ as much work as making the change, so I made the change. WDYT?